src/sentry/static/sentry/app/components/events/groupingInfo/groupingVariant.tsx

```diff
@@ -0,0 +1,183 @@
+    let component: EventGroupComponent | undefined;
+
+    if (variant.hash !== null) {
+      data.push([t('Hash'), variant.hash]);
+    }
+
+    if (variant.hashMismatch) {
+      data.push([
+        t('Hash mismatch'),
+        t('hashing algorithm produced a hash that does not match the event'),
+      ]);
+    }
+
+    switch (variant.type) {
+      case EventGroupVariantType.COMPONENT:
+        component = variant.component;
+        if (variant.config?.id) {
+          data.push([t('Grouping Config'), variant.config.id]);
+        }
+        break;
+      case EventGroupVariantType.CUSTOM_FINGERPRINT:
+        if (variant.values) {
+          data.push([t('Fingerprint values'), variant.values]);
+        }
+        break;
+      case EventGroupVariantType.SALTED_COMPONENT:
+        component = variant.component;
+        if (variant.values) {
+          data.push([t('Fingerprint values'), variant.values]);
+        }
+        if (variant.config?.id) {
+          data.push([t('Grouping Config'), variant.config.id]);
+        }
+        break;
+      default:
+        break;
+    }
+
+    if (component) {
+      data.push([
+        t('Grouping'),
+        <GroupingTree key={component.id}>
+          <GroupingComponent
+            component={component}
+            showNonContributing={this.state.showNonContributing}
+          />
+        </GroupingTree>,
+      ]);
+    }
+
+    return [data, component];
+  }
+
+  renderTitle() {
+    const {variant} = this.props;
+    const isContributing = variant.hash !== null;
+
+    return (
+      <Tooltip
+        title={isContributing ? t('Contributing variant') : t('Non-contributing variant')}
+      >
+        <GroupVariantTitle>
+          <ContributionIcon isContributing={isContributing} />
+          {t('By')}{' '}
+          {variant.description
+            .split(' ')
+            .map(i => capitalize(i))
+            .join(' ')}
+        </GroupVariantTitle>
+      </Tooltip>
+    );
+  }
+
+  renderContributionToggle() {
+    const {showNonContributing} = this.state;
+
+    return (
+      <ButtonBar merged active={showNonContributing ? 'all' : 'relevant'}>
+        <Button barId="relevant" size="xsmall" onClick={this.handleHideNonContributing}>
+          {t('Contributing values')}
+        </Button>
+        <Button barId="all" size="xsmall" onClick={this.handleShowNonContributing}>
+          {t('All values')}
+        </Button>
+      </ButtonBar>
+    );
+  }
+
+  render() {
+    const [data, component] = this.getVariantData();
+
+    return (
+      <GroupVariantListItem>
+        <Header>
+          {this.renderTitle()}
+          {hasNonContributingComponent(component) && this.renderContributionToggle()}
+        </Header>
+
+        <KeyValueList data={data} isContextData isSorted={false} />
+      </GroupVariantListItem>
+    );
+  }
+}
+
+const GroupVariantListItem = styled('li')`
+  margin-bottom: ${space(4)};
+  & + li {
```

Comment:
😬 Is there a way to do this without using a sibling selector?

Reply:
I replaced it with hr 👍